### ISSUE TITLE
OpenCover crashes with an InvalidOperationException after ~4h

### ISSUE THREAD

**Issue by pirluq** [OpenCover/opencover#962]
Please provide the following information when submitting an issue. 

> Where appropriate replace the `[ ]` with a `[X]`

### My Framework

- [ ] .NET 2
- [ ] .NET 3.5
- [ ] .NET 4
- [ ] .NET 4.5
- [ ] .NET 4.6
- [ ] .NET 4.6.1
- [ ] .NET 4.6.2
- [ ] .NET 4.7
- [ ] .NET 4.7.1
- [X] .NET 4.7.2
- [ ] .NET Core 1.0.0
- [ ] .NET Core 2.0.0
- [ ] Something else

### My Environment

- [ ] Windows 7 or below (not truly supported due to EOL)
- [ ] Windows 8
- [ ] Windows 8.1
- [ ] Windows 10
- [ ] Windows 10 IoT Core
- [X] Windows Server 2012
- [ ] Windows Server 2012 R2
- [ ] Windows Server 2016

### I have already...

- [X] repeated the problem using the latest stable release of OpenCover.
- [C] reviewed the [usage guide](https://github.com/opencover/opencover/wiki/usage) and [usage document](https://github.com/opencover/opencover/blob/master/main/OpenCover.Documentation/Usage.pdf).
- [X] have looked at the opencover output xml file in an attempt to resolve the issue.
- [X] reviewed the [current issues](https://github.com/opencover/opencover/issues) to check that the issue isn't already known.

### My issue is related to (check only those which apply):

- [X] no coverage being recorded
- [ ] 32 or 64 bit support
- [ ] feature request

### Expected Behavior

I've been trying to get the coverage from our solution for the past 3 months now.
I use the latest version of opencover 4.7.922.

It was working before, but with our project migration to sdk style (.net 4.7.2 and netstandard2 multi target), I'm a bit lost.

### Actual Behavior

I get this error log after about 4hours of run.
We have 8 millions lines of code to cover and ~300,000 unit tests to run.
the whole thing is running on a dedicated server with 512GB RAM)

2020-09-29 16:14:42,967 [1] FATAL OpenCover [(null)] - At: Program.Main
2020-09-29 16:14:43,096 [1] FATAL OpenCover [(null)] - An System.InvalidOperationException occurred: Operation is not valid due to the current state of the object.
2020-09-29 16:14:43,158 [1] FATAL OpenCover [(null)] - stack:    at OpenCover.Framework.Communication.MessageHandler.StandardMessage(MSG_Type msgType, IManagedCommunicationBlock mcb, Action`2 chunkReady, Action`1 offloadHandling) in C:\Repos\OpenCover\main\OpenCover.Framework\Communication\MessageHandler.cs:line 113
   at OpenCover.Framework.Communication.CommunicationManager.HandleCommunicationBlock(IManagedCommunicationBlock mcb, Action`1 offloadHandling) in C:\Repos\OpenCover\main\OpenCover.Framework\Communication\CommunicationManager.cs:line 60
   at OpenCover.Framework.Manager.ProfilerManager.ProcessMessages(WaitHandle[] handles) in C:\Repos\OpenCover\main\OpenCover.Framework\Manager\ProfilerManager.cs:line 241
   at OpenCover.Framework.Manager.ProfilerManager.RunProcess(Action`1 process, String[] servicePrincipal) in C:\Repos\OpenCover\main\OpenCover.Framework\Manager\ProfilerManager.cs:line 120
   at OpenCover.Console.Program.RunWithContainer(CommandLineParser parser, Bootstrapper container, IPersistance persistance) in C:\Repos\OpenCover\main\OpenCover.Console\Program.cs:line 127
   at OpenCover.Console.Program.Main(String[] args) in C:\Repos\OpenCover\main\OpenCover.Console\Program.cs:line 58
2020-09-29 16:14:43,161 [1] FATAL OpenCover [(null)] - A report has been sent to the OpenCover development team.
2020-09-29 16:14:43,163 [1] ERROR OpenCover [(null)] - If you are unable to resolve the issue please contact the OpenCover development team
2020-09-29 16:14:43,163 [1] ERROR OpenCover [(null)] - see https://www.github.com/opencover/opencover/issues

### Steps to reproduce the problem:

- [X] reviewed the [usage guide](https://github.com/opencover/opencover/wiki/usage) and [usage document](https://github.com/opencover/opencover/blob/master/main/OpenCover.Documentation/Usage.pdf) again because we know you probably didn't do it the first time.
- [ ] sample code or link to repository/gist is available (support is much faster this way)

**Comment by sawilde** [comment]
run your tests in smaller batches and merge the coverage results - you could even run them in parallel and thus speed up your build time

**Comment by trivalik** [comment]
I got this issue today also once, for an ancient project with old project style targeting .NET 4.0. with less than 80000 lines coverage on last successful run.

**Comment by sawilde** [comment]
Sorry to hear that - if you can repeat the issue and able to share it then you can raise a new issue, alternatively you can pull the code and have a go at debugging it yourself.
